pkg/registry/apis/provisioning/register.go

```diff
@@ -583,8 +583,15 @@ func (b *APIBuilder) GetPostStartHooks() (map[string]genericapiserver.PostStartH
 			commenter := pullrequest.NewCommenter()
 			pullRequestWorker := pullrequest.NewPullRequestWorker(evaluator, commenter)
 
-			driver := jobs.NewJobDriver(time.Second*28, time.Second*30, time.Second*30, b.jobs, b, b.jobHistory,
+			driver, err := jobs.NewJobDriver(
+				time.Minute*20, // Max time for each job
+				time.Minute*25, // Cleanup any checked out jobs
```

Comment:
that's pretty long time for the clean up :( a job could be stack for 25 minutes if one of the worker dies. 

We need to implement a logic for keep updating the claim. 

This would also affects in single tenant on restarts when it happens during the job.